### DISCUSSION TITLE
[Feature] Provide option to support auto-generate doc id by ES  for insert trace record when use elasticsearch storage

### DISCUSSION THREAD

**Discussion by samz-github** [apache/skywalking#12658] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

My corporation elasticsearch DBA feedback to me that skywalking use custom doc id butnot auto-generated by ES cluster, and it makes ES performace not good. 
I have checked storage-eleasticsearch-plugin source code.  Skywalking OAP query ES with  the doc id  for UI template. For query trace record, not using doc id, but other ES field, like segment_id or trace_id.  Can we not to generate custom doc id when insert  trace recod in ES? Instead, it just let ES auto generate doc id when inserting trace record.  

`

    @Override
    public InsertRequest prepareBatchInsert(Model model, Record record) throws IOException {
        final ElasticSearchConverter.ToStorage toStorage = new ElasticSearchConverter.ToStorage(model.getName());
        storageBuilder.entity2Storage(record, toStorage);
        Map<String, Object> builder = IndexController.INSTANCE.appendTableColumn(model, toStorage.obtain());
        String modelName = TimeSeriesUtils.writeIndexName(model, record.getTimeBucket());
        String id = IndexController.INSTANCE.generateDocId(model, record.id().build());
        Optional<String> routingValue = RoutingUtils.getRoutingValue(model, toStorage);
        return getClient().prepareInsert(modelName, id, routingValue, builder);
    }
`

### Use case

It helps to improve ES performace when  insert and search  ES trace records, especially for huge trace records everyday . 

### Are you willing to submit a pull request to implement this on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
This is intentional. We controlled the ID to make sure it would be updated by doc ID, rather than query condition. This improves the performance.

**Comment by samz-github** [comment]
Updated scene can be found in MetricsESDao class.   For trace record doc,  we just insert and query, and no update, right? If we mostly use trace record and seldom use metric, it's good to have a option to disable custom doc id?
